h/h_api/bulk_api/command_processor.py

```diff
@@ -104,20 +104,26 @@ def _check_command_count(self, final=False):
         :param final: This is the final count check, not incremental
         """
 
-        if not self.config:
```

Comment:
Had to swap the order here. Previously we did the in-line check first and the final check at the end of this block of code. This had the bug where we would not bother with the final check if no command length expectation had been set. 

We now fail with `CommandSequenceError` if there are no commands.